### PR TITLE
[ACS-5556] [ACS-5557] [ACS-5558] [ACS-5559] [ACS-5602] [ACS-5603] mat-datepicker component added and removed old mat-datetimepicker component from UI

### PR DESCRIPTION
**Pull request by jatin2008**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [x] Bugfix
> - [ ] Feature
> - [x] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
old mat-datetimePicker component is being used in the UI

**What is the new behaviour?**
Added new date-timepicker component and removed old mat-datetimePicker component

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
Jira id - https://alfresco.atlassian.net/browse/ACS-5557

Images for new datetime-picker

Screenshot from 2023-07-11 13-30-14

Screenshot from 2023-07-11 13-31-19

### CODE REVIEW

File: lib/core/src/lib/card-view/components/card-view-dateitem/card-view-dateitem.component.spec.ts [hunk 1]
```diff
@@ -194,6 +193,7 @@ describe('CardViewDateItemComponent', () => {
         const itemUpdatedSpy = spyOn(cardViewUpdateService.itemUpdated$, 'next');
         component.editable = true;
         component.property.editable = true;
+        component.property.value = moment('Jul 10 2017', 'MMM DD YYYY').endOf('day').toDate();
```
Comment (DenysVuika) [inline lib/core/src/lib/card-view/components/card-view-dateitem/card-view-dateitem.component.spec.ts:196]: can this be date-fns ?

File: lib/core/src/lib/card-view/components/card-view-dateitem/card-view-dateitem.component.spec.ts [hunk 2]
```diff
@@ -355,23 +355,4 @@ describe('CardViewDateItemComponent', () => {
         expect(valueChips[1].nativeElement.innerText.trim()).toBe('Jul 11, 2017');
         expect(valueChips[2].nativeElement.innerText.trim()).toBe('Jul 12, 2017');
     });
-
-    it('should render chips for multivalue datetimes when chips are enabled', async () => {
```
Comment (DenysVuika) [inline lib/core/src/lib/card-view/components/card-view-dateitem/card-view-dateitem.component.spec.ts:359]: why this test is removed? do you plan adding a replacement?

File: lib/core/src/lib/card-view/models/card-view-datetimeitem.model.ts [hunk 3]
```diff
@@ -22,7 +22,7 @@ import { CardViewDateItemProperties } from '../interfaces/card-view.interfaces';
 
 export class CardViewDatetimeItemModel extends CardViewDateItemModel implements CardViewItem, DynamicComponentModel {
     type: string = 'datetime';
-    format: string = 'MMM d, y, H:mm';
+    format: string = 'MMM d, y';
```
Comment (DenysVuika) [inline lib/core/src/lib/card-view/models/card-view-datetimeitem.model.ts:25]: the model file is called "datetime" model, and the "type" is called "datetime", so you can't drop the time from here

File: lib/core/src/lib/card-view/components/card-view-dateitem/card-view-dateitem.component.spec.ts (hunk 1 above)
Comment (jatin2008) [reply lib/core/src/lib/card-view/components/card-view-dateitem/card-view-dateitem.component.spec.ts]: yess converted to date-fns

File: lib/core/src/lib/card-view/models/card-view-datetimeitem.model.ts (hunk 3 above)
Comment (jatin2008) [reply lib/core/src/lib/card-view/models/card-view-datetimeitem.model.ts]: i have again implemented tim here

File: lib/core/src/lib/card-view/components/card-view-dateitem/card-view-dateitem.component.spec.ts [hunk 4]
```diff
@@ -314,25 +333,9 @@ describe('CardViewDateItemComponent', () => {
         });
     });
 
-    it('should be possible update a date-time', async () => {
```
Comment (nikita-web-ua) [inline lib/core/src/lib/card-view/components/card-view-dateitem/card-view-dateitem.component.spec.ts:317]: Why did you delete this test?

Comment (jatin2008) [reply lib/core/src/lib/card-view/components/card-view-dateitem/card-view-dateitem.component.spec.ts:317]: hi @nikita-web-ua i have not deleted this test, just modified it's name and added it's functionality in updateDateTIme() function to avoid duplication of code.

Comment (jatin2008) [reply lib/core/src/lib/card-view/components/card-view-dateitem/card-view-dateitem.component.spec.ts:317]: Because we have 2 cases here now one with `properties.cm:from` and other without `properties.cm:from`

**Review by rbahirsheth — APPROVED** [review]

File: lib/core/src/lib/card-view/components/card-view-dateitem/card-view-dateitem.component.spec.ts [hunk 5]
```diff
@@ -61,7 +61,25 @@ describe('CardViewDateItemComponent', () => {
...
+        const element = fixture.debugElement.nativeElement.querySelector(`span[data-automation-id="card-date-value-${key}"]`);
+        expect(element).toBeDefined();
+        expect(element.innerText).toEqual('Jul 10, 2017');
+        component.onDateChanged({ value: expectedDate });
+
+        fixture.detectChanges();
+        return expectedDate;
+    };
+
+    //we will enable this test when will change from moment to date-fns throughout the application
```
Comment (AleksanderSklorz) [inline lib/core/src/lib/card-view/components/card-view-dateitem/card-view-dateitem.component.spec.ts:80]: I'm wondering if we should raise jira for that or maybe correct it for now based on actual functionality instead of disabling that so then when functionality will be corrected then test can be corrected too. @MichalKinas what do you think?

File: lib/core/src/lib/card-view/components/card-view-dateitem/card-view-dateitem.component.ts [hunk 6]
```diff
@@ -57,36 +53,38 @@ export class CardViewDateItemComponent extends BaseCardView<CardViewDateItemMode
...
+    constructor(private dateAdapter: DateAdapter<DateFnsAdapter>,
                 private userPreferencesService: UserPreferencesService,
-                private appConfig: AppConfigService,
                 private clipboardService: ClipboardService,
-                private translateService: TranslationService) {
+                private translateService: TranslationService,
+                @Inject(MAT_DATE_FORMATS) private dateFormatConfig: MatDateFormats) {
         super();
-        this.dateFormat = this.appConfig.get('dateValues.defaultDateFormat');
+        // need to change this to app.config when will change from moment to date-fns throughout the application
```
Comment (AleksanderSklorz) [inline lib/core/src/lib/card-view/components/card-view-dateitem/card-view-dateitem.component.ts:70]: Do we have jira for this comment? We can forget about that without jira

**Comment by MateuszHY** [comment]
@jatin2008 @rbahirsheth did you check accessibility after changes introduced here?
Plus ... any additional tests needed here?

File: lib/core/src/lib/card-view/components/card-view-dateitem/card-view-dateitem.component.ts [hunk 7]
```diff
@@ -154,4 +152,8 @@ export class CardViewDateItemComponent extends BaseCardView<CardViewDateItemMode
     update() {
         this.cardViewUpdateService.update({ ...this.property } as CardViewDateItemModel, this.property.value);
     }
+
+    getDateValue(newDateValue): Date {
```
Comment (AleksanderSklorz) [inline lib/core/src/lib/card-view/components/card-view-dateitem/card-view-dateitem.component.ts:156]: missing type for newDateValue

**Comment by jatin2008** [comment]
> @jatin2008 @rbahirsheth did you check accessibility after changes introduced here? Plus ... any additional tests needed here?

hi @MateuszHY we have checked and not found any accessibility issue in the new datepicker component added and also added a new test case for the changes we made.

File: lib/core/src/lib/card-view/components/card-view-dateitem/card-view-dateitem.component.ts (hunk 7 above)
Comment (jatin2008) [reply lib/core/src/lib/card-view/components/card-view-dateitem/card-view-dateitem.component.ts:156]: we are calling our method from onDateChanged() & this method is triggering on the date change from UI. This method accepts the $event object & does not use type in this method too. We can use MatDatepickerInputEvent<any, any> type here. Is it fine?

File: lib/core/src/lib/card-view/components/card-view-dateitem/card-view-dateitem.component.spec.ts (hunk 5 above)
Comment (jatin2008) [reply lib/core/src/lib/card-view/components/card-view-dateitem/card-view-dateitem.component.spec.ts:80]: We can keep this test case with xit as of now. We can raise a new Jira ticket to address this test case once we replace moment.js in whole application with date-fns. What do you think?

Comment (MichalKinas) [reply lib/core/src/lib/card-view/components/card-view-dateitem/card-view-dateitem.component.spec.ts:80]: I think it would be best not to create a partial PR for such a change. I believe the best approach would be to go through the whole ADF and replace it everywhere and then adjust all affected tests at once instead of disabling some of them. This approach makes it easier to miss something during testing phase, during unit test and e2e fixes and most importantly such a change shouldn't be merged partially as it may be released without being complete. cc @AleksanderSklorz @MateuszHY

Comment (AleksanderSklorz) [reply lib/core/src/lib/card-view/components/card-view-dateitem/card-view-dateitem.component.spec.ts:80]: I agree with you @MichalKinas

Comment (MateuszHY) [reply lib/core/src/lib/card-view/components/card-view-dateitem/card-view-dateitem.component.spec.ts:80]: Agreed with @MichalKinas 
@jatin2008 @rbahirsheth what do you think ?

Comment (rbahirsheth) [reply lib/core/src/lib/card-view/components/card-view-dateitem/card-view-dateitem.component.spec.ts:80]: Hi, @MichalKinas I do agree with you. We found some occurrences in the codebase but we could not find out the UI. We will analyze it. But, This PR is also part of the calendar replacement.

File: lib/core/src/lib/card-view/components/card-view-dateitem/card-view-dateitem.component.ts (hunk 7 above)
Comment (jatin2008) [reply lib/core/src/lib/card-view/components/card-view-dateitem/card-view-dateitem.component.ts]: hi @AleksanderSklorz added type here as MatDatepickerInputEvent<any, any> for $event, also checked and found **any** as the best supported type here.

Comment (MichalKinas) [reply lib/core/src/lib/card-view/components/card-view-dateitem/card-view-dateitem.component.ts]: @jatin2008 based on what information you think that `any` is the best supported type? That's not true, please go through the documentation https://material.angular.io/components/datepicker/overview especially the section with `change event` to check what type should be used here

Comment (MichalKinas) [reply lib/core/src/lib/card-view/components/card-view-dateitem/card-view-dateitem.component.ts]: Btw `any` is against strongly typed TypeScript code, it's basically antipattern that we should avoid as much as we can, so please update your knowledge about best practices for TS

File: lib/core/src/lib/card-view/components/card-view-dateitem/card-view-dateitem.component.ts [hunk 8]
```diff
@@ -57,36 +53,38 @@ export class CardViewDateItemComponent extends BaseCardView<CardViewDateItemMode
...
     private onDestroy$ = new Subject<boolean>();
 
-    constructor(private dateAdapter: DateAdapter<Moment>,
+    constructor(private dateAdapter: DateAdapter<DateFnsAdapter>,
                 private userPreferencesService: UserPreferencesService,
-                private appConfig: AppConfigService,
                 private clipboardService: ClipboardService,
-                private translateService: TranslationService) {
+                private translateService: TranslationService,
+                @Inject(MAT_DATE_FORMATS) private dateFormatConfig: MatDateFormats) {
```
Comment (DenysVuika) [inline lib/core/src/lib/card-view/components/card-view-dateitem/card-view-dateitem.component.ts:68]: never ever do this, this is a singleton access to all the date time formats, changing it will change all components

**Comment by DenysVuika** [comment]
closing as abandoned and of poor quality
